### PR TITLE
ARM64: GenerateFastNeg

### PR DESCRIPTION
**Pull request by sigatrev**
update for 64bit tagging, reflecting the amd64 version

### CODE REVIEW

File: lib/Backend/arm64/LowerMD.cpp [hunk 1]
```diff
@@ -3810,10 +3811,20 @@ LowererMD::GenerateFastNeg(IR::Instr * instrNeg)
...
 
+    if (opndSrc1->IsEqual(instrNeg->GetDst()))
+    {
+        usingNewDst = true;
+        opndDst = IR::RegOpnd::New(TyInt32, this->m_func);
+    }
+    else
+    {
+        opndDst = instrNeg->GetDst()->UseWithNewType(TyInt32, this->m_func);
+    }
```
Comment (agarwal-sandeep) [inline lib/Backend/arm64/LowerMD.cpp]: nit: Do this after the IsIntConst if

File: lib/Backend/arm64/LowerMD.cpp [hunk 2]
```diff
@@ -3841,52 +3852,45 @@ LowererMD::GenerateFastNeg(IR::Instr * instrNeg)
...
     // dst = SUB zr, src
 
-    instr = IR::Instr::New(Js::OpCode::SUB, opndDst, IR::RegOpnd::New(nullptr, RegZR, TyMachReg, this->m_func), opndSrc1, this->m_func);
-    instrNeg->InsertBefore(instr);
-
-    // dst = ADD dst, 2
-
-    instr = IR::Instr::New(Js::OpCode::ADDS, opndDst, opndDst, IR::IntConstOpnd::New(2, TyMachReg, this->m_func), this->m_func);
+    instr = IR::Instr::New(Js::OpCode::SUB, opndDst, IR::RegOpnd::New(nullptr, RegZR, TyInt32, this->m_func), opndSrc1, this->m_func);
     instrNeg->InsertBefore(instr);
```
Comment (aaronsgiles) [inline lib/Backend/arm64/LowerMD.cpp]: Need to use SUBS if you want to test for overflow.

**Comment by sigatrev** [comment]
Updated

File: lib/Backend/arm64/LowerMD.cpp [hunk 3]
```diff
@@ -3841,52 +3843,55 @@ LowererMD::GenerateFastNeg(IR::Instr * instrNeg)
...
+        opndDst = opndDst->UseWithNewType(TyMachPtr, this->m_func);
+    }
+
+    GenerateInt32ToVarConversion(opndDst, instrNeg);
+
+    if (usingNewDst)
     {
-        //Now store the result.
-        LowererMD::CreateAssign(instrNeg->GetDst(), opndDst, instrNeg);
+        instr = IR::Instr::New(Js::OpCode::MOV, instrNeg->GetDst(), opndDst, this->m_func);
```
Comment (agarwal-sandeep) [inline lib/Backend/arm64/LowerMD.cpp]: >instr [](start = 8, length = 5)

Why removed  LowererMD::CreateAssign? that have size and writebarrier check which is good.

Comment (sigatrev) [reply lib/Backend/arm64/LowerMD.cpp]: I initially made it as similar as possible to the amd64 version. I can change this back to CreateAssign.
